### PR TITLE
update session and database prefix to sf3_

### PR DESCRIPTION
**Pull request by HeahDude**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | yes |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | - |
- [ ] Allow this BC break,
- [ ] Related PR : https://github.com/symfony/security-acl/pull/22, https://github.com/doctrine/DoctrineBundle/pull/513

### CODE REVIEW

**Comment by javiereguiluz** [comment]
Could we just use `sf` as the prefix instead of `sf2`, `sf3`, etc. to prevent this error from happening whenever we release a new major version?

**Comment by HeahDude** [comment]
@javiereguiluz indeed ;) see @fabpot comment here https://github.com/symfony/security-acl/pull/22

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/session.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
 
     <parameters>
-        <parameter key="session.metadata.storage_key">_sf2_meta</parameter>
+        <parameter key="session.metadata.storage_key">_sf3_meta</parameter>
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/session.xml]: This is a BC break for projects relying on the meta (as existing sessions use the old key)

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php [hunk 2]
```diff
@@ -106,7 +106,7 @@ private function addAclSection(ArrayNodeDefinition $rootNode)
                             ->addDefaultsIfNotSet()
                             ->children()
                                 ->scalarNode('id')->end()
-                                ->scalarNode('prefix')->defaultValue('sf2_acl_')->end()
+                                ->scalarNode('prefix')->defaultValue('sf3_acl_')->end()
```
Comment (stof) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php]: This is a BC break too, as any existing project will have sf2 in their database

**Comment by stof** [comment]
IMO, we cannot allow this BC break in a patch version

**Comment by fabpot** [comment]
I would say this is a won't fix. There is no point in dealing with BC here, so let's keep sf2, that does not change anything anyway.

**Comment by HeahDude** [comment]
fair enough

**Pull request closed without merging**